src/kbmod/search/raw_image_eigen.h

```diff
@@ -0,0 +1,250 @@
+#ifndef RAWIMAGEEIGEN_H_
+#define RAWIMAGEEIGEN_H_
+
+
+#include <vector>
+#include <fitsio.h>
+#include <float.h>
+#include <iostream>
+#include <stdexcept>
+#include <Eigen/Core>
+
+#include "common.h"
+#include "psf.h"
+#include "pydocs/raw_image_docs.h"
+
+
+namespace search {
+
+  struct Index {
+    unsigned i;
+    unsigned j;
+
+    Index(unsigned idx_i, unsigned idx_j)
+      : i(idx_i), j(idx_j) {}
+
+    Index(int x, int y)
+      : i(x), j(y) {}
+
+    Index(float x, float y)
+      : i(floor(x)), j(floor(y)) {}
+
+    Index(float x, float y, bool floor)
```

Comment:
I was a bit confused about what is the correct way to instantiate the class from values that have a contextual interpretation based on the context in which they are used. This is overloaded float constructor, one of which floors the float, the other type-casts. Not sure if this is C++ approved use-case of overloading.

Intended to be the same as

```
Index(float x, float y, floor=false){
    if (floor) return Index(floor(x), floor(y))
    return Index(static_cast<unsigned>(x), static_cast<unsigned>(y)
```

As mentioned above, in the `interpolate` method we `floor` values, but then use `get_pixel` which just casts the value and we don't seem to worry about that a whole lot.

```
// The iv[01, 3, 4, 9, 10] are floats. If I paste the get_pixel here it would read as:
// I added the (int) to point out what I think happens
float a = (iv[0] >= 0 && iv[0] < width && iv[1]) >= 0 && (iv[1] < height) ? image((int)iv[0], (int)iv[1]) : NO_DATA; 
```

Part of `Index` is basically me enumerating when things are cast and when they are rounded down. I don't think this particular situation is ok for a cast, but it basically doesn't fail because of how we set up our pixel coordinate to index map.